jep-012-01-raw-string-literals.md

```diff
@@ -0,0 +1,358 @@
+JMESPath grammar and helping to ensure consistency among implementations.
+
+This proposal seeks to add the following syntax to JMESPath:
+
+```
+'foobar'
+'foo\'bar'
+`bar` -> Parse error/warning (implementation specific)
+```
+
+## Motivation
+
+Raw string literals are provided in [various programming languages](https://en.wikipedia.org/wiki/String_literal#Raw_strings) in order to prevent
+language specific interpretation (i.e., JSON parsing) and remove the need for
+escaping, avoiding a common problem called [leaning toothpick syndrome (LTS)](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome). Leaning toothpick
+syndrome is an issue in which strings become unreadable due to excessive use of
+escape characters in order to avoid delimiter collision (e.g., `\\\\\\`).
+
+When evaluating a JMESPath expression, it is often necessary to utilize string
+literals that are not extracted from the data being evaluated, but rather
+statically part of the compiled JMESPath expression. String literals are useful
+in many areas, but most notably when invoking functions or building up
+multi-select lists and hashes.
+
+The following expression returns the number of characters found in the string
+`"foo"`. When parsing this expression, `` `"foo"` `` is parsed as a JSON value
+which produces the string literal value of `foo`:
+
+```
+`"foo"`
+```
+
+The following expression is functionally equivalent. Notice that the quotes are
+elided from the JSON literal:
+
+```
+`foo`
+```
+
+These string literals are parsed using a JSON parser according to
+[RFC 4627](https://www.ietf.org/rfc/rfc4627.txt), which will expand unicode
+escape sequences, newline characters, and several other escape sequences
+documented in RFC 4627 section 2.5.
+
+For example, the use of an escaped unicode value `\u002B` is expanded into
+`+` in the following JMESPath expression:
+
+```
+`"foo\u002B"` -> "foo+"
+```
+
+You can escape escape sequences in JSON literals to prevent an escape sequence
+from being expanded:
+
+```
+`"foo\\u002B"` -> "foo\u002B"
+`foo\\u002B` -> "foo\u002B"
+```
+
+While this allows you to provide literal strings, it presents the following
+problems:
+
+
+1. Incurs an additional JSON parsing penalty.
+
+
+2. Requires the cognitive overhead of escaping escape characters if you
+actually want the data to be represented as it was literally provided
+(which can lead to LTS). If the data being escaped was meant to be used
+along with another language that uses `\` as an escape character, then the
+number of backslash characters doubles.
+
+
+3. Introduces an ambiguous rule to the JMESPath grammar that requires a prose
+based specification to resolve the ambiguity in parser implementations.
+
+At the time of authoring the original version of this JEP, the relevant literal grammar rules were [then defined](https://github.com/jmespath-community/jmespath.py/commit/95a393e32e5aeb5f984cf2d7811fad75d4e0df95) as follows:
```

Comment:
With the above section header, this can be restored.
```suggestion
The relevant literal grammar rules are currently defined as follows:
```

Reply:
I’m not sure I follow. The grammar as written in JEP12a is not aligned with the current grammar.